exonum/src/runtime/mod.rs

```diff
@@ -227,85 +228,93 @@ impl From<RuntimeIdentifier> for u32 {
 /// the provided `ExecutionContext`. Discrepancy in node behavior within these methods may lead
 /// to a consensus failure.
 ///
-/// The other `Runtime` methods may execute logic specific to the node.
+/// Other `Runtime` methods may execute logic specific to the node.
 ///
 /// # Handling Panics
 ///
-/// A panic in the `Runtime` methods will **not** be caught
-/// and will cause node termination. You may use [`catch_panic`](error/fn.catch_panic.html) method
-/// to catch panics in the Rust code and convert them to unchecked execution errors.
+/// Panics in the `Runtime` methods are **not** caught. A panic in the runtime method will cause the node termination.
+/// To catch panics in the Rust code and convert them to unchecked execution errors, use the
+/// [`catch_panic`](error/fn.catch_panic.html) method.
 #[allow(unused_variables)]
 pub trait Runtime: Send + fmt::Debug + 'static {
     /// Initializes the runtime, providing a `Blockchain` instance for further use.
     ///
-    /// This method is guaranteed to be called before any other `Runtime` methods. It is
-    /// called *exactly once* during `Runtime` lifetime.
+    /// Calling this method always takes place before calling any other `Runtime` methods. The `initialize`
+    ///  method is called *exactly once* during the `Runtime` lifetime.
     ///
     /// The default implementation does nothing.
     fn initialize(&mut self, blockchain: &Blockchain) {}
 
-    /// Notifies the runtime that the dispatcher has completed re-initialization after
-    /// node restart, including restoring the deployed artifacts / started service instances
-    /// for all runtimes.
+    /// Notifies the runtime that the dispatcher has completed re-initialization after the
+    /// node restart. Re-initialization includes restoring the deployed artifacts / started service instances
+    /// for all the runtimes.
     ///
-    /// This method is called *no more than once* during `Runtime` lifetime. It is called iff
-    /// the blockchain had genesis block when the node was started. The blockchain state
-    /// is guaranteed to not change between `initialize` and `on_resume` calls.
+    /// This method is called *maximum once* during the `Runtime` lifetime. To call the method,
```

Comment:
IMO the new variant slightly shifts the main point: original line has a strong accent on the conditions, under which this method **will** be invoked, but the new line describes conditions in which this method **may** be invoked. Maybe something like "Blockchain will invoke the method iff it has a genesis block when the node is started"?

Reply:
@aleksuss this suggestion still holds